File: admin-web/src/api/appeal.ts
import { get, post } from './request'

export interface AppealListParams {
  enterpriseName?: string
  creditCode?: string
  status?: string
  appealTypeCode?: string
  regionCode?: string
  responsibleDeptId?: string
  startDate?: string
  endDate?: string
  pageNo?: number
  pageSize?: number
}

export interface AppealItem {
  id: number
  appealNo: string
  enterpriseId: number
  enterpriseName: string
  creditCode: string
  title: string
  content: string
  contactName: string
  contactPhone: string
  industryCode?: string
  industryName?: string
  regionCode: string
  regionName: string
  appealTypeCode?: string
  appealTypeName?: string
  urgencyLevel?: string
  status: string
  handleMode?: string
  responsibleDeptId?: string
  responsibleDeptName?: string
  replyDeadline?: string
  submittedAt?: string
  acceptedAt?: string
  repliedAt?: string
  completedAt?: string
  evaluatedAt?: string
  createdAt?: string
  updatedAt?: string
}

export interface AppealRecord {
  id: number
  actionType: string
  actionName: string
  beforeStatus: string | null
  afterStatus: string
  opinion: string | null
  operatorType: string
  operatorName: string
  operatorDeptName: string | null
  createdAt: string
}

export interface AppealAssignment {
  id: number
  assignedDeptId: string
  assignedDeptName: string
  assignOpinion: string | null
  assignedAt: string
  deadline: string | null
  status: string
  replyContent: string | null
  repliedAt: string | null
}

export interface AppealEvaluation {
  id: number
  satisfaction: string
  score: number
  resolvedFlag: number
  comment: string | null
  evaluateTime: string
}

export interface AppealFollowup {
  id: number
  followupStatus: string
  followupMethod: string | null
  followupContent: string | null
  followupResult: string | null
  followupUserName: string | null
  followupTime: string | null
}

export interface AppealAttachment {
  id: number
  originalName: string
  fileExt: string
  fileSize: number
  storagePath: string
}

export interface AppealDetail extends AppealItem {
  attachments: AppealAttachment[]
  records: AppealRecord[]
  assignments: AppealAssignment[]
  evaluation: AppealEvaluation | null
  followups: AppealFollowup[]
}

export interface PageResult<T> {
  records: T[]
  total: number
  pageNo: number
  pageSize: number
}

export function getAppealList(params: AppealListParams): Promise<PageResult<AppealItem>> {
  return get<PageResult<AppealItem>>('/api/admin/appeals', params as Record<string, unknown>)
}

export function getAppealDetail(id: number): Promise<AppealDetail> {
  return get<AppealDetail>(`/api/admin/appeals/${id}`)
}

export function acceptAppeal(id: number, body: {
  appealTypeCode?: string
  appealTypeName?: string
  replyDeadline?: string
  opinion?: string
}): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/accept`, body)
}

export function returnSupplementAppeal(id: number, body: { opinion: string }): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/return-supplement`, body)
}

export function rejectAppeal(id: number, body: {
  reasonCode?: string
  reasonName?: string
  opinion?: string
}): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/reject`, body)
}

export function centerHandleAppeal(id: number, body: {
  replyContent: string
  attachmentIds?: number[]
}): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/center-handle`, body)
}

export function assignAppeal(id: number, body: {
  assignedDeptId: string
  assignedDeptName: string
  deadline?: string
  assignOpinion?: string
}): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/assign`, body)
}

export function departmentReplyAppeal(id: number, body: {
  replyContent: string
  attachmentIds?: number[]
}): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/department-reply`, body)
}

export function reviewReplyAppeal(id: number, body: {
  pass: boolean
  opinion?: string
}): Promise<AppealItem> {
  return post<AppealItem>(`/api/admin/appeals/${id}/review-reply`, body)
}

export function addAppealFollowup(id: number, body: {
  responsibleDeptId?: string
  responsibleDeptName?: string
  followupMethod?: string
  followupContent?: string
  followupResult?: string
}): Promise<unknown> {
  return post(`/api/admin/appeals/${id}/followup`, body)
}

File: app/schemas/appeal.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List


# ── Enterprise-side requests ──────────────────────────────────────────────────

class AppealSubmitRequest(BaseModel):
    title: str = Field(..., max_length=200)
    content: str
    contactName: str = Field(..., max_length=100)
    contactPhone: str = Field(..., max_length=64)
    industryCode: Optional[str] = None
    industryName: Optional[str] = None
    regionCode: str
    regionName: str
    urgencyLevel: Optional[str] = None
    attachmentIds: Optional[List[int]] = None


class AppealModifyRequest(BaseModel):
    title: Optional[str] = Field(None, max_length=200)
    content: Optional[str] = None
    contactName: Optional[str] = Field(None, max_length=100)
    contactPhone: Optional[str] = Field(None, max_length=64)
    industryCode: Optional[str] = None
    industryName: Optional[str] = None
    regionCode: Optional[str] = None
    regionName: Optional[str] = None
    urgencyLevel: Optional[str] = None


class AppealSupplementRequest(BaseModel):
    content: str
    attachmentIds: Optional[List[int]] = None


class AppealEvaluationRequest(BaseModel):
    satisfaction: str
    score: int = Field(..., ge=1, le=5)
    resolvedFlag: Optional[int] = None
    comment: Optional[str] = Field(None, max_length=1000)


# ── Admin-side requests ───────────────────────────────────────────────────────

class AppealAcceptRequest(BaseModel):
    appealTypeCode: str
    appealTypeName: str
    replyDeadline: Optional[datetime] = None
    opinion: Optional[str] = None


class AppealReturnSupplementRequest(BaseModel):
    opinion: str


class AppealRejectRequest(BaseModel):
    reasonCode: Optional[str] = None
    reasonName: Optional[str] = None
    opinion: str


class AppealCenterHandleRequest(BaseModel):
    replyContent: str
    attachmentIds: Optional[List[int]] = None


class AppealAssignRequest(BaseModel):
    assignedDeptId: str
    assignedDeptName: str
    deadline: Optional[datetime] = None
    assignOpinion: Optional[str] = None


class AppealDeptReplyRequest(BaseModel):
    replyContent: str
    attachmentIds: Optional[List[int]] = None


class AppealReviewReplyRequest(BaseModel):
    passed: bool = Field(..., alias="pass")
    opinion: Optional[str] = None
    model_config = {"populate_by_name": True}


class AppealFollowupRequest(BaseModel):
    responsibleDeptId: Optional[str] = None
    responsibleDeptName: Optional[str] = None
    followupMethod: Optional[str] = None
    followupContent: Optional[str] = None
    followupResult: Optional[str] = None


class AppealCompleteRequest(BaseModel):
    remark: Optional[str] = None


# ── Query params ──────────────────────────────────────────────────────────────

class EnterpriseAppealListQuery(BaseModel):
    status: Optional[str] = None
    pageNo: int = 1
    pageSize: int = 10


class AdminAppealListQuery(BaseModel):
    enterpriseName: Optional[str] = None
    creditCode: Optional[str] = None
    status: Optional[str] = None
    appealTypeCode: Optional[str] = None
    regionCode: Optional[str] = None
    responsibleDeptId: Optional[str] = None
    startDate: Optional[str] = None
    endDate: Optional[str] = None
    pageNo: int = 1
    pageSize: int = 10
